airflow/providers/amazon/aws/sensors/eks.py

```diff
@@ -0,0 +1,132 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""Tracking the state of EKS Clusters and Nodegroups."""
+
+from typing import Optional
+
+from boto3 import Session
+
+from airflow.providers.amazon.aws.hooks.eks import EKSHook
+from airflow.sensors.base import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+CONN_ID = "eks"
+REGION = Session().region_name
```

Comment:
Does this have the effect of making a request to the metaserver when loading the file? It seems to me that this value can also be overwritten by the connection configuration. https://github.com/apache/airflow/blob/2625007c8aeca9ed98dea361ba13c2622482d71f/airflow/providers/amazon/aws/hooks/base_aws.py#L83

Reply:
Upon further digging, it looks like we can set the default here to None and let the base hook do its magic if the user does not provide one.  I will try that and see what happens.

Note to self:  Whatever we do here, also apply the same solution to the Operators.

Reply:
Corrected in coming revision.

Reply:
This should be addressed in https://github.com/apache/airflow/pull/16571/commits/5726815e0e07376449d3938a15f31fc4cbca162a